munit/shared/src/main/scala/munit/Assertions.scala

```diff
@@ -116,6 +116,28 @@ trait Assertions extends MacroCompat.CompileErrorMacro {
   )(implicit loc: Location, ev: B <:< A): Unit = {
     StackTraces.dropInside {
       if (obtained != expected) {
+
+        (obtained, expected) match {
+          case (a: Array[_], b: Array[_]) if a.sameElements(b) =>
```

Comment:
That seems a bit puzzling to fail an assertion and ask the user to call the same assertion with converting `Array` -> `Seq`, although we have already done the job of checking elements... But I agree, it may help in understanding what's going on for those who forgot about `Array`'s equals checking. 

Shouldn't we add an analogical warning to `assertNotEquals` for a successful case (arrays are not equal by refs, but same on elements)?

Reply:
I don’t think it’s puzzling. It’s just an error message in the style of “did you mean?” compile errors.

You can already achieve the assertNotEquals behavior with an assert. The main point of the custom assertions is to provide richer error messages for common usecases. You can also enrich the assert failure message with clues.